operator/beiboot/handler/clusters.py

```diff
@@ -0,0 +1,105 @@
+from datetime import datetime
+
+import kubernetes as k8s
+import kopf
+
+from beiboot.clusterstate import BeibootCluster
+from beiboot.configuration import configuration
+
+core_api = k8s.client.CoreV1Api()
+objects_api = k8s.client.CustomObjectsApi()
+
+
+def _in_any_beiboot_namespace(event, namespace, kind: list, **_):
+    beiboots = objects_api.list_namespaced_custom_object(
+        group="getdeck.dev",
+        version="v1",
+        namespace=configuration.NAMESPACE,
+        plural="beiboots",
+    )
+    namespaces = set(bbt.get("beibootNamespace") for bbt in beiboots["items"])
+    return namespace in namespaces and event["object"]["involvedObject"]["kind"] in kind
+
+
+def workloads_in_beiboot_namespace(event, namespace, **_):
+    return _in_any_beiboot_namespace(
+        event, namespace, kind=["StatefulSet", "Deployment", "Pod"]
+    )
+
+
+def get_beiboot_for_namespace(namespace: str):
+    class AttrDict(dict):
+        def __init__(self, *args, **kwargs):
+            super(AttrDict, self).__init__(*args, **kwargs)
+            self.__dict__ = self
+
+    beiboots = objects_api.list_namespaced_custom_object(
+        group="getdeck.dev",
+        version="v1",
+        namespace=configuration.NAMESPACE,
+        plural="beiboots",
+    )
+    for bbt in beiboots["items"]:
+        if bbt["beibootNamespace"] == namespace:
+            # need to wrap this for correct later use
+            return AttrDict(bbt)
+    else:
+        return None
+
+
+@kopf.on.event(
+    "event",
+    when=kopf.all_(
+        [lambda event, **_: event["type"] is not None, workloads_in_beiboot_namespace]
```

Comment:
Just style - why one lambda and one def - maybe both def is a bit more consistent. Took me a while to understand.